App viewer tester

brombomb opened the pull request:
Check it out, it's deployed live already (based on this branch):

https://tronbyt.github.io/apps/app-viewer/index.html

The github workflow will update the pages, anytime code is merged to main.

Happy to answer any questions or make any other changes

brombomb commented on app-viewer/main.js:
```diff
@@ -0,0 +1,286 @@
+    // App detail page
+    renderAppDetail();
+  }
+});
```
implemented

brombomb commented on app-viewer/main.js:
```diff
@@ -0,0 +1,286 @@
+  }
+
+  // Clear container and create elements programmatically
+  container.innerHTML = '';
```
implemented

tavdog commented:
Ok that looks good.  One thing I notice is if the app has no details available then you can't get to the report broken button.

tavdog commented:
Also, anytime I click details they all just say "not documentation available".   The name and summary etc.  should be populated from the manifest file first and extra info pulled from an existing readme file.  Most apps don't have a readme.

brombomb commented:
I don't actually have a tron/tidybit I was trying to see the apps before I bought one, hence this MR.  

I changed the main page and details to list the info from the manifest first, since it looks like most apps provide that.  Then if there is a README.md it gets added to the App Details page.  This also means almost all apps have a Details button to get to the Report Broken button.  

I was trying to clean up the workflow, and took this branch out of the regenerate, then made some changes.  Added that back in so new changes should rebuild.  I see some errors at the moment, so I will fix those a bit later

brombomb commented:
> Wdyt about rendering the images with the dot-matrix overlay to match the style of tronbyt-server?

Is this what you're talking about?
<img width="963" height="512" alt="image" src="https://github.com/user-attachments/assets/577bbbf7-9234-41e5-9a3c-2bbb0c794d1c" />

brombomb commented:
> > Is this what you're talking about?
> 
> Yes, exactly!

It's a bit funky as some images already render using a dot appearance, so I added a toggle, to choose to see it on and off.

tavdog commented:
This looks fine.  We do already have an add app page with a search function for the server but I do like this as a way for people to see the apps before setting up a server for themselves.

tavdog commented:
I saw the last build failed for this but I didn't know why.  Might just be a spurious failure.
